tsdb/engine/tsm1/encoding.go

```diff
@@ -86,7 +86,45 @@ func (a Values) Size() int {
 	return sz
 }
 
-func (a Values) Filter(min, max int64) Values {
+// Merge overlays b to top of a.  If two values conflict with
+// the same timestamp, b is used.  Both a and b must be sorted
+// in ascending order.
+func (a Values) Merge(b Values) Values {
+	if len(a) == 0 {
+		return b
+	}
+
+	if len(b) == 0 {
+		return a
+	}
+
+	var i, j int
+	for i < len(a) && j < len(b) {
+		if a[i].UnixNano() < b[j].UnixNano() {
+			i++
+		} else if a[i].UnixNano() > b[j].UnixNano() {
+			a[i], b[j] = b[j], a[i]
+			i++
+		} else {
+			a[i] = b[j]
+			i++
+			j++
+		}
```

Comment:
Minor nit but we should probably save `UnixNano()` since we're calling it twice. Maybe `atime` & `btime`?